rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -141,42 +142,41 @@ void FileSourceReader::reset() {
 }
 
 Reader::Status FileSourceReader::subfolder_reading() {
-    if ((_sub_dir = opendir(_folder_path.c_str())) == nullptr)
-        THROW("FileReader ShardID [" + TOSTR(_shard_id) + "] ERROR: Failed opening the directory at " + _folder_path);
-
     std::vector<std::string> entry_name_list;
-    std::string _full_path = _folder_path;
-
-    while ((_entity = readdir(_sub_dir)) != nullptr) {
-        std::string entry_name(_entity->d_name);
-        if (strcmp(_entity->d_name, ".") == 0 || strcmp(_entity->d_name, "..") == 0) continue;
```

Comment:
Is this handled?

Reply:
Please let me know how we have handled this.

Reply:
The filesys::recursive_directory_iterator API stops when it has iterated through all files and directories in the specified directory and its sub directories. Hence we need not check if it exists / no

Hence we need not explicitly add this condition